src/dag_combine.rs

```diff
@@ -0,0 +1,56 @@
+use digraph::PyDiGraph;
+use pyo3::prelude::*;
+use petgraph::algo;
+use pyo3::Python;
+use pyo3::types::PyAny;
+use std::cmp::Ordering;
+
+fn contains_weight(py: Python, dag: &PyDiGraph, obj: &PyObject) -> i32 {
+    let mut index :i32 = -1;
+
+    for node in dag.graph.node_indices() {
+        let weight = dag.graph.node_weight(node).unwrap();
+        if ! weight.is_none(py) {
+
+            let predicate_callable = |a: &PyAny, b: &PyAny| -> PyResult<bool> {
+                let res = a.compare(b)?;
+                Ok(res == Ordering::Equal)
+            };
+
+            if predicate_callable(obj.as_ref(py), weight.as_ref(py)).unwrap() {
```

Comment:
Instead of unwrapping here you can just use `?` to return the Error if there is one:

```suggestion
            if predicate_callable(obj.as_ref(py), weight.as_ref(py))? {
```